llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -217,6 +219,15 @@ std::string mangleFuncItanium(StringRef BaseName, const FunctionType *FT) {
       (Twine("_Z") + Twine(BaseName.size()) + Twine(BaseName)).str();
   for (unsigned I = 0; I < FT->getNumParams(); ++I)
     Res += manglePrimitiveType(FT->getParamType(I));
+  if (FT->getReturnType()->isArrayTy() ||
```

Comment:
Nit: ideally, it would be good to be able to demangle the mangled intrinsic name. With growing mangling/demangling complexity existing Itanium mangler might be another cheap alternative. Is it worth a TODO?

Reply:
> ideally, it would be good to be able to demangle the mangled intrinsic name. 

I'm not sure if that is achievable, though, considering that we do have situations when we call two `__spirv_` functions which are only different by their return type; i.e. we explicitly introducing a return type into the mangled name, which I think won't be ever compatible with existing standard manglers/demanglers

Reply:
Added a `FIXME` in c88f6ffc433b324c3e3359d4887cc1c027ff2c64